### ISSUE TITLE
Github context is not properly referenced within the JSON payload template

### ISSUE THREAD

**Issue by ylkhayat** [slackapi/slack-github-action#84]
### Description
Github context is not properly referenced when using the `workflow_run`

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [x] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Create a `workflow_run` github action

#### Reproducible in:
package version: v1.18.0
node version: v16.13.0
OS version(s):

#### Steps to reproduce:
1. create a Github Action running on the `workflow_run` events
2. add a json template that uses `${{github.event.workflow_run.name}}` for example
3. fire the event, and the json template will not properly replace the github context

Example of my JSON template
```json
{
  "text": "[Codebase News] - Pipeline is 🔴",
  "attachments": [
    {
      "color": "#000",
      "blocks": [
        {
          "type": "section",
          "text": {
            "type": "mrkdwn",
            "text": "Pipeline is 🔴 on `master`."
          }
        },
        {
          "type": "section",
          "text": {
            "type": "mrkdwn",
            "text": "Workflow `${{github.event.workflow_run.name}}` requires extra attention. :warning:"
          }
        },
        {
          "type": "context",
          "elements": [
            {
              "type": "mrkdwn",
              "text": "More info here ${{github.server_url}}/${{github.repository}}/actions/runs/${{github.github.event.workflow_run.run_id}}"
            }
          ]
        }
      ]
    }
  ]
}

```

#### Expected result:

I would expect for the JSON file to have in hand the entire provided github context corresponding to the `workflow_run` event triggers.

#### Actual result:
As you can see `${{github.event.workflow_run.name}}` did not work as expected
<img width="570" alt="image" src="https://user-images.githubusercontent.com/29442110/164945714-92ce4d05-bf4f-41ee-afcf-605b33aae95a.png">
Also the regular github context did not work as well, here
<img width="618" alt="image" src="https://user-images.githubusercontent.com/29442110/164945746-e0dd725c-6db2-4549-b58c-0797d25d3d1b.png">

### Side Node:
I looked into this issue https://github.com/slackapi/slack-github-action/issues/51, and was aware that the problem got fixed, but I tested it multiple times, it was not properly fixed.

**Comment by FredericoIRodrigues** [comment]
Currently facing the same problem.

**Comment by nitintejuja** [comment]
@yousseftarekkh i was able to see github context variables message data in my slack message . 
i have done following changes in payload json 
`
{
  "text": "[Codebase News] - Pipeline is 🔴",
  "attachments": [
    {
      "color": "#000",
      "blocks": [
        {
          "type": "section",
          "text": {
            "type": "mrkdwn",
            "text": "Pipeline is 🔴 on `master`."
          }
        },
        {
          "type": "section",
          "text": {
            "type": "mrkdwn",
            "text": "Workflow `${{ github.event.workflow_run.name }}` requires extra attention. :warning:"
          }
        },
        {
          "type": "context",
          "elements": [
            {
              "type": "mrkdwn",
              "text": "More info here ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.event.workflow_run.run_id }}"
            }
          ]
        }
      ]
    }
  ]
}
`


i have received below message in slack 
![image](https://user-images.githubusercontent.com/95347924/172013529-6e1d2893-7ab1-4f64-b211-c61b42c42eb8.png)
regarding `github.event.workflow_run.name` that is bug .

**Comment by taofang2021** [comment]
When using a payload template, slack-action gets the `github.context` from [actions/toolkit](https://github.com/actions/toolkit/blob/c5278cdd088a8ed6a87dbd5c80d7c1ae03beb6e5/packages/github/src/context.ts#L10-L24) and r[eplaces the variables](https://github.com/slackapi/slack-github-action/blob/main/src/slack-send.js#L40-L42) in the template.
You will find that there is no `github.event.*` and `github.repository` in the toolkit, you can get `github.serverUrl` but not `github.server_url`.

**Comment by magnattic** [comment]
I have the same problem with the following snippet in the payload file:

```
        {
          "type": "button",
          "text": {
            "type": "plain_text",
            "text": "▶️ Check running actions"
          },
          "style": "primary",
          "url": "https://github.com/my-company/platform/actions?query=branch%3A${{ env.BRANCH_NAME }}"
        },
        {
          "type": "button",
          "text": {
            "type": "plain_text",
            "text": "📑 Open PR"
          },
          "url": "https://github.com/my-company/platform/pull/${{ steps.head-ref-match.outputs.group1 }}"
        },
```

Neither my custom env variable in `{{ env.BRANCH_NAME }}` nor referencing another step output with `${{ steps.head-ref-match.outputs.group1 }}` seems to work.

Any workaround for this?

**Comment by ivorsmorenburg** [comment]
This bug its horrible please release a bugfix 
![image](https://user-images.githubusercontent.com/48212434/199988421-efc52986-9e5f-4a0d-9068-7a6098534821.png)

The most basic and important thing of this app (sending messages with github info) not working

**Comment by ivorsmorenburg** [comment]
This makes me want to switch off from Slack

**Comment by jr-surfline** [comment]
I am seeing the same issue here. The payload-file-path template does not replace the env vars in the JSON template

Workflow code
```
      - name: Workflow init slack message
        id: slack-init-msg
        uses: slackapi/slack-github-action@v1.23.0
        env:
          STATUS: "Success"
          BRANCH: "WPT-000_Test_stuff"
          APP_NAME: "My_sweet_app"
          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
          WORKFLOW_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
        with:
          channel-id: "my-test-channel"
          payload-file-path: ./.github/slack-msg-payloads/init-msg.json
```

Template JSON file
```
{
    "blocks": [
    {
        "type": "header",
        "text": {
        "type": "plain_text",
        "text": "Deployment of ${{ env.APP_NAME }} to ${{ env.ENVIRO }} started",
        "emoji": true
        }
    },
    {
        "type": "context",
        "elements": [
        {
            "type": "mrkdwn",
            "text": "*Workflow URL:* `${{ env.WORKFLOW_URL }}`\n*BRANCH:* `${{ env.BRANCH }}`\n*Triggered by:* `${{ github.actor }}`"
        }
        ]
    },
    {
        "type": "divider"
    }
    ],
    "attachments": [
    {
        "color": "#0328fc",
        "blocks": [
        {
            "type": "section",
            "text": {
            "type": "mrkdwn",
            "text": "This area will be updated with the workflow progress, stay tuned!\n*Workflow status:* ${{ env.STATUS }}"
            }
        }
        ]
    }
    ]
}
```
Output in Slack
![Screen Shot 2022-12-07 at 8 23 57 PM](https://user-images.githubusercontent.com/87091336/206356199-a4dfb2d4-f343-4f8a-acff-be165c915f23.png)

**Comment by rd-florian-stagliano** [comment]
any news on this? experiencing the same issue using a slack webhook and a json payload in a template file

**Comment by filmaj** [comment]
We fixed this in #200; assigning to the 1.24 milestone, which we will release shortly.

**Comment by filmaj** [comment]
Release v1.24.0 is out!
